docs/components/guidesXml.rst

```diff
@@ -0,0 +1,71 @@
+
+..  _guides-xml:
+
+==========
+guides.xml
+==========
+
+..  todo: Add general docs about guides.xml
+
+..  _guides-xml-add-config-extension:
+
+Add a configuration option (Extension)
+======================================
+
+Extension authors can add additional configuration options in the
+extension's tag:
+
+..  code-block:: xml
+    :caption: guides.xml
+
+    <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension"
+               project-home="https://docs.typo3.org/"
+               project-contact="https://typo3.slack.com/archives/C028JEPJL"/>
+
+These must then be interpreted by the extension itself. The interpretation
+is usually done in a :php:class:`phpDocumentor\Guides\Event\PostProjectNodeCreated`
+event.
+
+..  _guides-xml-add-config-core:
+
+Add a configuration option (Contributors, directly in the guides)
+=================================================================
+
+In order to add a configuration option to the :file:`guides.xml` the option
+must be added to the XML schema of the :file:`guides.xml`. You can find
+it at :file:`packages/guides-cli/resources/schema/guides.xsd` in the
+mono repository.
+
+Register the new configuration option in the :php:`TreeBuilder` at
+:php:func:`phpDocumentor\Guides\DependencyInjection\GuidesExtension::getConfigTreeBuilder`.
+
+..  todo: link symfony docs
+
+Using the tree builder is a topic of itself. Refer to the according symfony documentation.
+
+If the configuration is a setting option you can save the value in class
```

Comment:
I must admit I don't know what this would mean, maybe an example could clear up what is referred to?

Reply:
good point, adding some examples